x-pack/plugins/security_solution/cypress/e2e/detection_rules/prebuilt_rules_install_update_workflows.cy.ts

```diff
@@ -0,0 +1,193 @@
+      rule_id: 'rule_2',
+    });
+    beforeEach(() => {
+      preventPrebuiltRulesPackageInstallation();
+
+      /* Create two mock rules */
+      createNewRuleAsset({ rule: RULE_1 });
+      createNewRuleAsset({ rule: RULE_2 });
+      waitForRulesTableToBeLoaded();
+    });
+
+    it('should install selected rules when user clicks on Install selected rules', () => {
+      addElasticRulessButtonClick();
+      cy.get(SELECT_ALL_RULES_ON_PAGE_CHECKBOX).click();
+      cy.get(INSTALL_SELECTED_RULES_BUTTON).click();
+      cy.get(TOASTER).should('be.visible').should('have.text', `2 rules installed successfully.`);
+      cy.get(GO_BACK_TO_RULES_TABLE_BUTTON).click();
+      cy.get(RULES_MANAGEMENT_TABLE).find(RULES_ROW).should('have.length', 2);
+      cy.get(RULES_MANAGEMENT_TABLE).contains(RULE_1['security-rule'].name);
+      cy.get(RULES_MANAGEMENT_TABLE).contains(RULE_2['security-rule'].name);
+    });
+
+    it('should fail gracefully with toast error message when request to install rules fails', () => {
+      /* Stub request to force rules installation to fail */
+      cy.intercept('POST', '/internal/detection_engine/prebuilt_rules/installation/_perform', {
+        statusCode: 500,
+      }).as('installPrebuiltRules');
+      addElasticRulessButtonClick();
+      cy.get(SELECT_ALL_RULES_ON_PAGE_CHECKBOX).click();
+      cy.get(INSTALL_SELECTED_RULES_BUTTON).click();
+      cy.wait('@installPrebuiltRules');
+      cy.get(TOASTER).should('be.visible').should('have.text', 'Rule installation failed');
+    });
+  });
+
+  describe('Update of prebuilt rules', () => {
+    const RULE_ID = 'rule_id';
+    const OUTDATED_RULE = createRuleAssetSavedObject({
+      name: 'Outdated rule',
+      rule_id: RULE_ID,
+      version: 1,
+    });
+    const UPDATED_RULE = createRuleAssetSavedObject({
+      name: 'Updated rule',
+      rule_id: RULE_ID,
+      version: 2,
+    });
+    beforeEach(() => {
+      preventPrebuiltRulesPackageInstallation();
+      /* Create a new rule and install it */
+      createNewRuleAsset({ rule: OUTDATED_RULE });
+      installAllPrebuiltRulesRequest();
+      /* Create a second version of the rule, making it available for update */
+      createNewRuleAsset({ rule: UPDATED_RULE });
+      waitForRulesTableToBeLoaded();
+      reload();
+    });
+
+    it('should update rule succesfully', () => {
+      cy.intercept('POST', '/internal/detection_engine/prebuilt_rules/upgrade/_perform').as(
+        'updatePrebuiltRules'
+      );
+      ruleUpdatesTabClick();
+      cy.get(RULES_UPDATES_TABLE).find(RULES_ROW).should('have.length', 1);
+      cy.get(RULES_UPDATES_TABLE).contains(OUTDATED_RULE['security-rule'].name);
+      cy.get(UPGRADE_ALL_RULES_BUTTON).click();
+      cy.wait('@updatePrebuiltRules');
+      cy.get(TOASTER).should('be.visible').should('have.text', `1 rule updated successfully.`);
+    });
+
+    it('should fail gracefully with toast error message when request to update rules fails', () => {
```

Comment:
there are a lot of repeating commands in both tests. Can they be moved to separate helpers?

Reply:
Makes sense. Refactored to helper.